src/neural/cuda/network_cuda.cc

```diff
@@ -76,7 +76,10 @@ static size_t getMaxAttentionHeadSize(const LegacyWeights& weights, int N) {
   const size_t output_size = N * (64 * 64 + 8 * 24);
 
   size = std::max(size, std::max(matmul_qk_size, output_size));
-  return size;
+
+  // We process the k, q, and v matrix multiplications in parallel using single allocation
+  // so need 3x of this.
+  return size * 3;
```

Comment:
This seems pessimistic - shouldn't you multiply encoder_d_model by 3 instead and policy_d_model by 2?

Reply:
yeah, will make it more accurate.